vm/src/frame.rs

```diff
@@ -806,30 +804,39 @@ impl Frame {
         module: &str,
         symbol: &Option<String>,
     ) -> FrameResult {
-        let current_path = {
-            let mut source_pathbuf = PathBuf::from(&self.code.source_path);
-            source_pathbuf.pop();
-            source_pathbuf
+        let import = vm.builtins.get_item("__import__");
+        let module = match import {
+            Some(func) => vm.invoke(func, vec![vm.ctx.new_str(module.to_string())])?,
+            None => panic!("No __import__ in builtins"),
         };
 
-        let obj = import(vm, current_path, module, symbol)?;
+        // If we're importing a symbol, look it up and use it, otherwise construct a module and return
+        // that
+        let obj = match symbol {
+            Some(symbol) => module.get_attr(symbol).map_or_else(
+                || {
+                    let import_error = vm.context().exceptions.import_error.clone();
+                    Err(vm.new_exception(import_error, format!("cannot import name '{}'", symbol)))
+                },
+                Ok,
+            ),
+            None => Ok(module),
+        };
 
         // Push module on stack:
-        self.push_value(obj);
+        self.push_value(obj?);
         Ok(None)
     }
 
     fn import_star(&self, vm: &mut VirtualMachine, module: &str) -> FrameResult {
-        let current_path = {
-            let mut source_pathbuf = PathBuf::from(&self.code.source_path);
-            source_pathbuf.pop();
-            source_pathbuf
+        let import = vm.builtins.get_item("__import__");
+        let module = match import {
+            Some(func) => vm.invoke(func, vec![vm.ctx.new_str(module.to_string())])?,
```

Comment:
I would suggest to add a new method to the `VirtualMachine` class, called `import`, which will lookup the builtin, and invoke the import function. All other functions should call this function to import a module.